File: ohjisu/SWEA/PYTHON/swea2056.py
"""
2056. 연월일 달력 D1

"YYYY/MM/DD"형식으로 출력하고,

날짜가 유효하지 않을 경우, -1 을 출력하는 프로그램을 작성하라.
연월일로 구성된 입력에서 월은 1~12 사이 값을 가져야 하며

일은 [표1] 과 같이, 1일 ~ 각각의 달에 해당하는 날짜까지의 값을 가질 수 있다.

"""



T = int(input())
for tc in range(1, T+1) :
    result = ''
    date = input()
    # 출력할 형식 맞추기
    result = date[:4] + '/' + date[4:6] + '/' + date[6:]
    # 조건을 넣을 dictionary 만들기
    condition = {}
    # month 리스트 만들기
    month = ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12']
    # update dictionary
    for key in month :
        if int(key) % 2 == 1 :                  # 홀수 달일 때, 1~31
            condition.update({key: range(1, 32)}) 
        elif key == '02' :                      # 2월은 1~28
            condition.update({key: range(1, 29)})
        else :                                  # 짝수 달일 때, 1~30
            condition.update({key: range(1, 31)})
    if  (date[4:6] not in month)  or (int(date[6:]) not in condition[date[4:6]]) : # 조건에 해당하지 않을 때는
        result = -1                                                                # result에 -1 할당
    print(f'#{tc} {result}')

File: aeongseo/SWEA/CPP/swea2056.cpp
/*** 2056. 연월일 달력 (D1) ***/
/*
연월일 순으로 구성된 8자리의 날짜가 입력으로 주어진다.
해달 날짜의 유호ㅛ성을 판단한 후, 날짜가 유효하다면 "YYYY/MM/DD"형식으로 출력하고, 날짜가 유효하지 않을 경우, -1을 출력하는 프로그램을 작성하라.
*/

#include<iostream>
#include<string>
using namespace std;

int main(){
    int T;
    cin >> T;

    for (int tc=1; tc<T+1; ++tc){
        string date;
        cin >> date;

        // substr() 함수 : 문자열의 특정 부분 잘라냄. s.substr(시작위치, 길이)
        string s_year = date.substr(0, 4);
        string s_month = date.substr(4, 2);
        string s_day = date.substr(6);

        int result;
        // stoi() 함수 : sring 입력받고, 출력으로 integer 반환
        int month = stoi(s_month);
        int day = stoi(s_day);

        // 31일까지인 달 : 1, 3, 5, 7, 8, 10, 12
        if ((month==1 || month==3 || month==5 || month==7 || month==8 || month==10 || month==12) && (day>=1 && day<=31)){
            cout << '#' << tc << ' ' << s_year << '/' << s_month << '/' << s_day << '\n';
        }
        // 30일까지인 달 : 4, 6, 9, 11
        else if ((month==4 || month==6 || month==9 || month==11) && (day>=1 && day<=30)) {
            cout << '#' << tc << ' ' << s_year << '/' << s_month << '/' << s_day << '\n';
        }

        // 28일까지인 달 : 2
        else if ((month == 2) && (day>=1 && day<=28)) {
            cout << '#' << tc << ' ' << s_year << '/' << s_month << '/' << s_day << '\n';
        }
        
        else {
            cout << '#' << tc << ' ' << -1 << '\n';
        }
        
    }

    return 0;
}